clients/src/test/java/org/apache/kafka/common/utils/TimeTest.java

```diff
@@ -26,9 +26,11 @@
 import static org.junit.jupiter.api.Assertions.assertNull;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
-public abstract class TimeTest {
```

Comment:
Is there a reason this needed to become concrete? I think the existing pattern was fine, and made it clear that the suite was parameterized by the createTime() method.

Reply:
Before I delete `SystimeTime`  and `SystemTimeTest` class,  which resulted in the class having no abstract method, so I changed it to concrete
Now I have restored it